scrapy/utils/misc.py

```diff
@@ -160,6 +160,50 @@ def create_instance(objcls, settings, crawler, *args, **kwargs):
     return instance
 
 
+def build_from_crawler(objcls, crawler, *args, **kwargs):
+    """
+    ``*args`` and ``**kwargs`` are forwarded to the constructors.
+    Raises ``ValueError`` if``crawler`` is``None``.
+    Raises typeError is instance is None
+    Creates a class instance using 'from_crawler' constructor
+    """
```

Comment:
Both new methods should get a proper complete docstring.

Reply:
Thank you for the feedback, the issues should be fixed now. I didn't have additional documentations to add for the methods, but I reformatted them similarly to other docstrings.